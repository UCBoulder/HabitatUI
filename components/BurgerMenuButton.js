import React from 'react'
import { TouchableOpacity, StyleSheet } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'
import PropTypes from 'prop-types'

export const BurgerMenuButton = ({ navigation }) => {
  return (
    <TouchableOpacity
      style={styles.burgerMenuButton}
      onPress={() => navigation.openDrawer()}
    >
      <Icon name="menu" size={30} color="black" />
    </TouchableOpacity>
  )
}

BurgerMenuButton.propTypes = {
  navigation: PropTypes.object
}

const styles = StyleSheet.create({
  burgerMenuButton: {
    position: 'absolute',
    top: 10,
    left: 10,
    padding: 10,
    borderRadius: 15,
    backgroundColor: 'white'
  }
})
