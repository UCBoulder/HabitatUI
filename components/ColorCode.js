export const ColorCode = (rating) => {
  // No=0 Maybe=1 Yes=2 increasing level of assurity
  if (rating === '2') {
    return '#031cfc'
  } else if (rating === '0') {
    return '#f7a202'
  } else if (rating === '1') {
    return '#f3f702'
  } else {
    return null
  }
}
