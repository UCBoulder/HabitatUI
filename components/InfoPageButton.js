import React from 'react'
import { TouchableOpacity, Text, View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

export const InfoPageButton = ({ navigation, destinationScreen, title }) => {
  return (
        <View style={styles.SwitchPageButton}>
            <TouchableOpacity
                style={styles.button}
                onPress={() => navigation.navigate(destinationScreen)}
            >
                <Text style={styles.buttonText}>{title}</Text>
            </TouchableOpacity>
        </View>
  )
}

InfoPageButton.propTypes = {
  navigation: PropTypes.object.isRequired,
  destinationScreen: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

const styles = StyleSheet.create({
  SwitchPageButton: {
    position: 'absolute',
    alignItems: 'center'
  },
  button: {
    backgroundColor: 'white',
    padding: 10,
    borderRadius: 5,
    marginLeft: 12,
    marginTop: 12
  },
  buttonText: {
    color: 'black'
  }
})
