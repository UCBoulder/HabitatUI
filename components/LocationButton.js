import React from "react";
import { View, Button, StyleSheet } from "react-native";

// button that appears on the map page
const LocationButton = ({ onGetLocation }) => (
  <View style={styles.LocationButton}>
    <Button title="Get user location" onPress={onGetLocation} />
  </View>
);

const styles = StyleSheet.create({
  LocationButton: {
    position: "absolute",
    bottom: 20,
    left: 0,
    right: 0,
    alignItems: "center",
  },
});

export default LocationButton;
