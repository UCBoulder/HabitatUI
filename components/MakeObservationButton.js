import React from 'react'
import { View, TouchableOpacity, Text, StyleSheet } from 'react-native'
import { useNavigation } from '@react-navigation/native'

// Button that appears on the map screen which starts the cheatgrass observation process for the user
const MakeObservationButton = () => {
  const navigation = useNavigation()

  const handlePress = () => {
    navigation.navigate('Camera')
  }

  return (
    <View style={styles.locationButtonContainer}>
      <TouchableOpacity
        style={styles.locationButton}
        onPress={handlePress}
      >
        <Text style={styles.buttonText}>Make Observation</Text>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  locationButtonContainer: {
    position: 'absolute',
    bottom: 20,
    left: 0,
    right: 0,
    alignItems: 'center'
  },
  locationButton: {
    backgroundColor: 'white',
    padding: 15,
    borderRadius: 10
  },
  buttonText: {
    color: 'black',
    textAlign: 'center',
    fontSize: 18
  }
})

export default MakeObservationButton
