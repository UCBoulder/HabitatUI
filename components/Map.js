import React from "react";
import { View, StyleSheet } from "react-native";
import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps";

const Map = ({ userLocation }) => (
  <View style={styles.mapContainer}>
    <MapView
      provider={PROVIDER_GOOGLE}
      style={styles.map}
      showsUserLocation={true}
      showsMyLocationButton={true}
      
      initialRegion={{
        latitude: 38.5449,
        longitude: -106.9329,
        latitudeDelta: 0.0922,
        longitudeDelta: 0.0421,
      }}
    >
      {userLocation && (
        <Marker
          coordinate={{
            latitude: userLocation.coords.latitude,
            longitude: userLocation.coords.longitude,
          }}
        />
      )}
    </MapView>
  </View>
);

const styles = StyleSheet.create({
  mapContainer: {
    minHeight: '100%'
  },
  map: {
    ...StyleSheet.absoluteFillObject,
  },
});

export default Map;
