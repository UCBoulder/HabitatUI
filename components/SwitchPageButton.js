import React from "react";
import { TouchableOpacity, Text, View, StyleSheet } from "react-native";

export const SwitchPageButton = ({ navigation, destinationScreen, title }) => {
    return (
        <View style={styles.SwitchPageButton}>
            <TouchableOpacity
                style={styles.button}
                onPress={() => navigation.navigate(destinationScreen)}
            >
                <Text style={styles.buttonText}>{title}</Text>
            </TouchableOpacity>
        </View>
    );
};

const styles = StyleSheet.create({
    SwitchPageButton: {
        position: "absolute",
        alignItems: "center",
    },
    button: {
        backgroundColor: "#fff",
        padding: 10,
        borderRadius: 0,
    },
    buttonText: {
        color: "#000",
    },
});
