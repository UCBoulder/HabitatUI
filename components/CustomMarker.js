import React, { useState } from 'react'
import { Marker, Callout } from 'react-native-maps'
import { ColorCode } from './ColorCode'
import { FormatDate } from '../utils/FormatDate'
import { View, Text, StyleSheet, Image, Modal, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import { simplifyJson } from '../utils/simplifyJson'
import Icon from 'react-native-vector-icons/FontAwesome6'
import FetchS3Image from '../utils/FetchS3Image'

const CustomMarker = ({ data }) => {
  const [modalVisible, setModalVisible] = useState(false)
  const parsedData = simplifyJson(data)

  const calloutPress = () => {
    setModalVisible(true)
  }

  const closeModal = () => {
    setModalVisible(false)
  }

  return (
    <Marker coordinate={{
      latitude: parsedData.coords.latitude,
      longitude: parsedData.coords.longitude
    }}
      pinColor={ColorCode(parsedData.VerificationRating ? parsedData.VerificationRating : '1')}>
      <Callout tooltip onPress={calloutPress}>

        <View style={styles.calloutContainer}>

          <Text style={styles.calloutText}>
            {`Observation made on: ${FormatDate(parsedData.timestamp)}\n`}
            {`Latitude: ${parsedData.coords.latitude}\nLongitude: ${parsedData.coords.longitude}\n`}
            {`Accuracy: ${parsedData.coords.accuracy}\n`}
            {`Notes: ${parsedData.Notes?.locationDescription ?? ''}\n`}
          </Text>

          <Text style={styles.calloutTextCentered}>
            {'Tap to view image'}
          </Text>

        </View>

        {/* Image popup */}
        <Modal
          animationType="slide"
          visible={modalVisible}
        >
          <View style={styles.modalContainer}>
            {/* <FetchS3Image /> */}
            {/* <Image
              source={parsedData.observationImageUrl }
              style={styles.modalImage}
              resizeMode="stretch"
            /> */}
            <TouchableOpacity style={styles.modalExitButton} onPress={closeModal}>
              <Icon name="x" size={25} color="white" />
            </TouchableOpacity>
          </View>

        </Modal>

      </Callout>
    </Marker>
  )
}

CustomMarker.propTypes = {
  data: PropTypes.shape({
    coords: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      accuracy: PropTypes.number
    }),
    VerificationRating: PropTypes.object,
    timestamp: PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.number
    ]),
    Notes: PropTypes.shape({
      estimatedCover: PropTypes.string,
      estimatedArea: PropTypes.string,
      locationDescription: PropTypes.string,
      ownership: PropTypes.string
    })
  })
}

const styles = StyleSheet.create({
  calloutContainer: {
    borderRadius: 10,
    backgroundColor: 'white',
    padding: 4
  },
  calloutText: {
    fontSize: 14,
    color: 'black'
  },
  calloutTextCentered: {
    fontSize: 14,
    color: 'black',
    textAlign: 'center'
  },
  modalContainer: {
    flex: 1,
    width: '100%',
    height: '100%'
  },
  modalExitButton: {
    zIndex: 2,
    position: 'absolute',
    top: 10,
    right: 10,
    borderRadius: 50,
    padding: 10
  },
  modalImage: {
    flex: 1,
    width: '100%',
    height: '100%'
  }
})

export default CustomMarker
