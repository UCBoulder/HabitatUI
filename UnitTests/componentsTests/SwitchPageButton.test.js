import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableOpacity, Text } from 'react-native';
import { SwitchPageButton } from '../../components/SwitchPageButton';

// Test if the button renders correcly and handles being pressed correctly
describe('SwitchPageButton', () => {
  it('renders correctly', () => {

    // Mock Props
    const navigation = { navigate: jest.fn() };
    const destinationScreen = 'Info';
    const title = 'Info';

    // Render the component
    const tree = renderer
      .create(<SwitchPageButton navigation={navigation} destinationScreen={destinationScreen} title={title} />)
      .toJSON();

    // Expect the rendered component to match the snapshot
    expect(tree).toMatchSnapshot();
  });

  it('handles press correctly', () => {
    // Mock any necessary props
    const navigation = { navigate: jest.fn() };
    const destinationScreen = 'Info';
    const title = 'Info';

    // Render the component
    const component = renderer.create(<SwitchPageButton navigation={navigation} destinationScreen={destinationScreen} title={title} />);
    const tree = component.toJSON();

    // Find the TouchableOpacity element and simulate a press
    const touchableOpacity = component.root.findByType(TouchableOpacity);
    touchableOpacity.props.onPress();

    // Expect the navigate function to be called with the correct argument
    expect(navigation.navigate).toHaveBeenCalledWith(destinationScreen);
  });
});
