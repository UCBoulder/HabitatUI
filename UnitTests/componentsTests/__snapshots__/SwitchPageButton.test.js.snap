// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchPageButton renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "position": "absolute",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      Info
    </Text>
  </View>
</View>
`;
